File: eliza/packages/python/elizaos/agent_orchestrator_env.py
"""Parity with TypeScript `plugin-collector`: optional agent orchestrator (PTY) via env."""

from __future__ import annotations

import os


def eliza_agent_orchestrator_load_requested() -> bool:
    """Match `elizaAgentOrchestratorLoadRequested()` in `plugin-collector.ts`."""
    raw = (os.environ.get("ELIZA_AGENT_ORCHESTRATOR") or "").strip().lower()
    if raw in ("0", "false", "no"):
        return False
    return raw in ("1", "true", "yes")

File: eliza/packages/rust/src/agent_orchestrator_env.rs
//! Parity with TypeScript `plugin-collector`: optional agent orchestrator (PTY) via
//! `ELIZA_AGENT_ORCHESTRATOR`.

/// Match `elizaAgentOrchestratorLoadRequested()` in `plugin-collector.ts`.
pub fn eliza_agent_orchestrator_load_requested() -> bool {
    match std::env::var("ELIZA_AGENT_ORCHESTRATOR") {
        Ok(raw) => {
            let normalized = raw.trim().to_lowercase();
            if matches!(normalized.as_str(), "0" | "false" | "no") {
                return false;
            }
            matches!(normalized.as_str(), "1" | "true" | "yes")
        }
        Err(_) => false,
    }
}
